### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Arvind-Srinivasan**
Fixed some install and documentation.

### CODE REVIEW

File: tests/cpp/leds.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "mtrain.hpp"
```
Comment (kasohrab) [inline tests/cpp/leds.cpp:1]: the header file is still called mtrain.hpp so it doesn't compile on my computer cause of this.

File: external/middleware/FreeRTOS/Source/include/FreeRTOSConfig.h [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+//  extern uint32_t SystemCoreClock;
+// #endif
+
+/*  CMSIS-RTOSv2 defines 56 levels of priorities. To be able to use them
+ *  all and avoid application misbehavior, configUSE_PORT_OPTIMISED_TASK_SELECTION
+ *  must be set to 0 and configMAX_PRIORITIES to 56
+ *
+ */
+#define configUSE_PORT_OPTIMISED_TASK_SELECTION	1
+/* #define configMAX_PRIORITIES					( 56 ) */
```
Comment (kasohrab) [inline external/middleware/FreeRTOS/Source/include/FreeRTOSConfig.h:55]:
```suggestion
```
we can delete this right?

**Review by kasohrab — CHANGES_REQUESTED** [review]
It's a lot to review but looking through it briefly I have some questions:
Do we need to delete all these c api files? If they work why can't we just keep them, just in case? 
Also are we sure the FatFs stuff doesn't work either?
And this has been tested on a board? If not then it's ok we can just wait until we can do that.

**Comment by Arvind-Srinivasan** [comment]
It's better to encourage a consistent API and language target. We had a lot of issues before where it was incredibly confusing to figure out why the compiler failed to compile some apis or worse yet, why code worked with one API and not the other.

**Comment by Arvind-Srinivasan** [comment]
The code has been tested by me but it would be good if someone else can revalidate all the tests

**Comment by kasohrab** [comment]
moving to monorepo so I already included this pr in that merge.

**Pull request closed without merging**